### PR TITLE
[EOS-21294] Validate devices , interfaces and health , manifest

### PR DESCRIPTION
**Pull request by mazinamdar**
Signed-off-by: mazinamdar <mazhar.inamdar@seagate.com>

1. Validate cvg devices.
2. Validate network interfaces.
3. Validate server type.
4. Validate health , Manefist.
5. Validate certificate installation.

### CODE REVIEW

File: lr-cli/cortx_setup/commands/node/finalize.py [hunk 1]
```diff
@@ -29,9 +34,86 @@ class NodeFinalize(Command):
...
+        if not cvgs or cvgs is MISSED:
+            raise CortxSetupError(f"Devices are not provided")
+        for cvg in cvgs:
+            meta_data = cvg.get('metadata_devices')
+            data = cvg.get('data_devices')
+            if not meta_data or not data:
+                raise CortxSetupError(f"metadata or data devices are missing")
+
+            for meta in meta_data:
+                if meta in data:
```
Comment (83bhp) [inline lr-cli/cortx_setup/commands/node/finalize.py:98]: Do we not need similar check check for data in meta_data?

File: lr-cli/cortx_setup/validate.py [hunk 2]
```diff
@@ -39,3 +41,26 @@ def path(path):
...
+        try:
+            cmd_run(f"ip a | grep {iface}")
+        except Exception as exc:
+            raise CortxSetupError(f"Invalid interface {iface}\n {exc}")
+
+
+def disk_devices(device_type, devices):
+    local_devices = None
+    if device_type == HW_TYPE:
+        local_devices = cmd_run("multipath -ll|grep mpath|sort -k2|cut -d' ' -f1|sed 's|mpath|/dev/disk/by-id/dm-name-mpath|g'|paste -s -d, -")  # noqa: E501
```
Comment (83bhp) [inline lr-cli/cortx_setup/validate.py:57]: will this fail if there are no multipath devices?

**Review by 83bhp — COMMENTED** [review]
couple of comments. 
Also, do we not need to validate the data from confstore?

File: lr-cli/cortx_setup/commands/node/finalize.py (hunk 1 above)
Comment (mazinamdar) [reply lr-cli/cortx_setup/commands/node/finalize.py:98]: I think only checking meta in data will raise error for duplicate devices vice versa is not required.

File: lr-cli/cortx_setup/validate.py (hunk 2 above)
Comment (mazinamdar) [reply lr-cli/cortx_setup/validate.py:57]: this command will return empty string, I have handle that case in code.

Comment (83bhp) [reply lr-cli/cortx_setup/validate.py:57]: ok..

**Review by 83bhp — APPROVED** [review]

**Pull request merged**